src/main/java/com/bathanh/exercise/basic/string/Ex303.java

```diff
@@ -0,0 +1,24 @@
+package com.bathanh.exercise.basic.string;
+
+public class Ex303 {
+
+    public int sumAllNumberString(final String s) {
+        int sum = 0;
+        int number = 0;
+        int j = 0;
+        for (int i = 0; i < s.length(); i++) {
+            if (s.charAt(i) >= '0' && s.charAt(i) <= '9') {
+                number = s.charAt(i) - '0';
+                for (j = i + 1; j < s.length(); j++) {
+                    if (s.charAt(j) < '0' || s.charAt(j) > '9') {
+                        break;
+                    }
+                    number = number * 10 + s.charAt(j) - '0';
+                }
+                sum += number;
+                i = j;
```

Comment:
i là biến vòng for, đừng cố gắng thay đổi giá trị của nó bằng bất kỳ 1 cách nào khác

Reply:
e cho i = j, ở đây sau khi lặp hết 1 dãy các số có trong chuỗi để bước tiếp theo vòng lặp i bỏ qua nhưng số đã xét này.

Reply:
vậy e không được làm cách này à a

Reply:
Em thử tối ưu lại một tí